src/bytes.rs

```diff
@@ -628,6 +647,39 @@ impl Regex {
         Ok(unsafe { match_data.find(&self.code, subject, start, options)? })
     }
 
+    /// Returns the same as is_partial_match, but starts the search at the given
+    /// offset.
+    ///
+    /// The significance of the starting point is that it takes the surrounding
+    /// context into consideration. For example, the `\A` anchor can only
+    /// match when `start == 0`.
+    pub fn is_partial_match_at(
+        &self,
+        subject: &[u8],
+        start: usize,
+    ) -> Result<bool, Error> {
+        assert!(
+            start <= subject.len(),
+            "start ({}) must be <= subject.len() ({})",
+            start,
+            subject.len()
+        );
+
+        let mut options = PCRE2_PARTIAL_HARD;
+        if !self.config.utf_check {
+            options |= PCRE2_NO_UTF_CHECK;
+        }
+
+        let match_data = self.match_data();
+        let mut match_data = match_data.borrow_mut();
+        // SAFETY: The only unsafe PCRE2 option we potentially use here is
+        // PCRE2_NO_UTF_CHECK, and that only occurs if the caller executes the
+        // `disable_utf_check` method, which propagates the safety contract to
+        // the caller.
+        Ok(unsafe { match_data.find(&self.code, subject, start, options)? })
+    }
```

Comment:
It looks to me like this code might benefit from a slight refactor. Namely, `is_match_at` and `is_partial_match_at` appear to only differ in implementation by a single thing: whether `PCRE2_PARTIAL_HARD` is set or not. A little code repetition doesn't normally bother me, but in this case, we can reduce the number of `unsafe` usages by 1 since I don't believe partial matching has any relevance to safety.

So I'd say, create a private `is_match_at_imp` routine with `options` as a parameter, and then call it from each of these functions with the appropriate initial condition.

Reply:
Ok, will do.